politeiad/config.go

```diff
@@ -410,16 +411,44 @@ func loadConfig() (*config, []string, error) {
 	}
 
 	// Validate cache options.
+	if cfg.EnableCache {
+		switch {
+		case cfg.CacheHost == "":
+			return nil, nil, fmt.Errorf("the enablecache param can " +
+				"not be used without the cachehost param")
+		case cfg.CacheRootCert == "":
+			return nil, nil, fmt.Errorf("the enablecache param can " +
+				"not be used without the cacherootcert param")
+		case cfg.CacheCert == "":
+			return nil, nil, fmt.Errorf("the enablecache param can " +
+				"not be used without the cachecert param")
+		case cfg.CacheKey == "":
+			return nil, nil, fmt.Errorf("the enablecache param can " +
+				"not be used without the cachekey param")
+		}
+	}
+
 	if cfg.BuildCache && !cfg.EnableCache {
-		str := "%s: The buildcache param can't be used " +
-			"without the enablecache param"
-		err := fmt.Errorf(str, funcName)
-		fmt.Fprintln(os.Stderr, err)
-		fmt.Fprintln(os.Stderr, usageMessage)
-		os.Exit(0)
+		return nil, nil, fmt.Errorf("the buildcache param can " +
+			"not be used without the enablecache param")
 	}
-	cfg.CacheCertDir = cleanAndExpandPath(cfg.CacheCertDir)
+
 	cfg.CacheRootCert = cleanAndExpandPath(cfg.CacheRootCert)
+	cfg.CacheCert = cleanAndExpandPath(cfg.CacheCert)
+	cfg.CacheKey = cleanAndExpandPath(cfg.CacheKey)
+
```

Comment:
A check that tls.LoadX509KeyPair() does not fail is needed.  Then two os.Stat() checks below can be removed.  A better check than just existing on the root cert is needed as well.

Reply:
Fixed.  The key pair is checked using tls.LoadX509KeyPair() and the root cert is checked using x509.ParseCertificate().